plugins/modules/sonic_image_management.py

```diff
@@ -197,7 +229,17 @@ def validate_and_retrieve_params(module, warnings):
             module.fail_json(msg="{0} -> name is required when {0} -> command = {1}".format(params['category'], params['command']))
         if params.get('path'):
             warnings.append("{0} -> path is ignored when {0} -> command = {1}".format(params['category'], params['command']))
-
+    elif params['command'] == 'gpg-key':
+        if not params.get('keyserver') or not params.get('pubkeyid'):
+            module.fail_json(msg="{0} -> keyserver URL and Key ID are required when {0} -> command = {1}".format(params['category'], params['command']))
+    elif params['command'] == 'verify':
+        if params.get('verifymethod') == 'gpg':
```

Comment:
Please check if 'verifymethod' is specified when the command is 'verify'
```suggestion
    elif params['command'] == 'verify':
        if not params.get('verifymethod'):
            module.fail_json(msg="{0} -> verifymethod is required when {0} -> command = verify".format(params['category']))
        if params.get('verifymethod') == 'gpg':
```

Reply:
Addressed.